x-pack/plugins/snapshot_restore/public/plugin.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { PLUGIN } from '../common/constants';
+import { CLIENT_BASE_PATH, renderReact } from './app';
+import template from './index.html';
+import { createShim } from './shim';
+
+const REACT_ROOT_ID = 'snapshotRestoreReactRoot';
+
+export class Plugin {
+  public start(): void {
+    const shim = createShim();
```

Comment:
Like in my comment on the server, I think you should pull the shim logic out of the `Plugin` class and into the code that instantiates this and calls .start(). Keep the new code as pristine as possible, and let the legacy code handle the hackery.